src/main/java/com/klarna/hiverunner/HiveServerContainer.java

```diff
@@ -149,6 +154,52 @@ public String apply(@Nullable Object[] objects) {
         }
     }
 
+    /*
+    There's an obscure bug in HiveRunner where if you have a View query with a Join, it will fail to execute the query
+    if the table names after the join are not lowercase, so this method transforms the query so that the parts
+    of the query that are case insensitive are lowercase
+    */
+    public String transformQuery(String hiveql) {
+        String result = hiveql;
+        int indexOn = hiveql.toLowerCase().indexOf(" on ");
+        if(hiveql.contains("'")){
+            //First we leave the query untouched from whatever is before the on
+            result = hiveql.substring(0, indexOn);
+            int indexFirstComa = hiveql.indexOf("'");
+            String subString = hiveql.substring(indexOn, hiveql.length());
+            //Count how many case sensitive strings there is
+            int count = StringUtils.countMatches(subString, "'");
+            int indexStart = subString.indexOf("'");
+
+            for(int i = 0; i < count/2; i++) {
```

Comment:
Could there ever be an uneven number of quote marks? 🤔  and does hive only allow single quotes `'`, and not double quotes `"`